app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt

```diff
@@ -180,7 +181,7 @@ class WireActivityViewModel @Inject constructor(
     }
 
     fun handleDeepLink(intent: Intent?) {
-        if (isSharingIntent(intent)) {
+        if (!shouldGoToWelcome() && isSharingIntent(intent)) {
```

Comment:
maybe another private function that would explain what those two conditions mean together ?